src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -83,7 +85,7 @@ internal sealed class RSession : IRSession, IRCallbacks {
             CanceledBeginInteractionTask = TaskUtilities.CreateCanceled<IRSessionInteraction>(new RHostDisconnectedException());
         }
 
-        public RSession(int id, string name, IBrokerClient brokerClient, IExclusiveReaderLock initializationLock, Action onDispose) {
+        public RSession(int id, string name, IBrokerClient brokerClient, IExclusiveReaderLock initializationLock, Action onDispose, IServiceContainer services) {
```

Comment:
`RSession` shouldn't have access to the `IServiceContainer` directly, it should interact with external world only through the callback. Why do you need it?